Graph/Traversal/BreadthFirstSearch.cpp

```diff
@@ -0,0 +1,57 @@
+/* Breadth-first search/traversal
+ * iterates through all vertices in an undirected graph in breadth-first order
+ *
+ * Time-complexity: 
+ * O(V + E), where V is number of vertices and E is number of edges
+ *
+ * Space-complexity:
+ * O(V)
+ *
+ */
+
+#include <queue>
+#include "BreadthFirstSearch.h"
+
+void Graph::breadthFirstSearch(int source_vertex) {
```

Comment:
`breadthFirstSearch` doesn't need to be member function of `Graph` class.

It should take as input a `Graph` object, along with `source_vertex` (which should be `const`).